stories/demos/exportUtils.tsx

```diff
@@ -0,0 +1,87 @@
+import { cloneElement } from 'react';
+import type { ReactElement } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import XLSX from 'xlsx';
+import { jsPDF } from 'jspdf';
+import autoTable from 'jspdf-autotable';
+
+import type { DataGridProps } from '../../src';
+
+export function exportToCsv<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const { head, body, foot } = getGridContent(gridElement);
+  const content = [...head, ...body, ...foot]
+    .map(cells => cells.map(serialiseCellValue).join(','))
+    .join('\n');
+
+  downloadFile(fileName, new Blob([content], { type: 'text/csv;charset=utf-8;' }));
+}
+
+export function exportToXlsx<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const { head, body, foot } = getGridContent(gridElement);
+  const wb = XLSX.utils.book_new();
+  const ws = XLSX.utils.aoa_to_sheet([...head, ...body, ...foot]);
+  XLSX.utils.book_append_sheet(wb, ws, 'Sheet 1');
+  XLSX.writeFile(wb, fileName);
+}
+
+
+export function exportToPdf<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const doc = new jsPDF({
+    orientation: 'l',
+    unit: 'px'
+  });
+
+  const { head, body, foot } = getGridContent(gridElement);
+  autoTable(doc, {
+    head,
+    body,
+    foot,
+    horizontalPageBreak: true,
+    styles: { cellPadding: 1.5, fontSize: 8, cellWidth: 'wrap' },
+    tableWidth: 'wrap'
+  });
+  doc.save(fileName);
+}
+
+function getGridContent<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>) {
+  const grid = document.createElement('content');
+  grid.innerHTML = renderToStaticMarkup(cloneElement(gridElement, {
+    enableVirtualization: false
+  }));
+
+  return {
+    head: getRows('.rdg-header-row'),
+    body: getRows('.rdg-row:not(.rdg-summary-row)'),
+    foot: getRows('.rdg-summary-row')
+  };
+
+  function getRows(selector: string) {
+    return Array.from(
+      grid.querySelectorAll<HTMLDivElement>(selector)
+    ).map(gridRow => {
+      return Array.from(
+        gridRow.querySelectorAll<HTMLDivElement>('.rdg-cell')
+      ).map(gridCell => gridCell.innerText);
```

Comment:
So the cell ends up empty when we display an image, a checkbox, or a `progress` element for example?

Reply:
Correct. We can manually update cell value during export or provide a flag to the formatter so it can return an print friendly value